applications/tari_merge_mining_proxy/src/main.rs

```diff
@@ -388,6 +424,50 @@ fn get_spending_key() -> (PrivateKey, PrivateKey) {
     (key, r)
 }
 
+// Select lowest difficulty
+fn select_difficulty(data: &[u8], tari_difficulty: u64) -> Vec<u8> {
+    // Parse the JSON
+    let parsed = json::parse(&stringify_request(data));
+    match parsed {
+        Ok(mut parsed) => {
+            if parsed["result"]["difficulty"].is_null() {
+                error!(
+                    target: LOG_TARGET,
+                    "{:#}",
+                    MmProxyError::ParseError("Monero response invalid, cannot determine difficulty to use".to_string())
+                );
+                return data.to_vec();
+            }
+            let monero_difficulty = &parsed["result"]["difficulty"].as_u64();
+            match monero_difficulty {
+                Some(monero_difficulty) => {
+                    let difficulty = min(monero_difficulty, &tari_difficulty).to_owned();
+                    parsed["result"]["difficulty"] = difficulty.into();
+                    parsed.dump().into()
+                },
+                None => {
+                    error!(
+                        target: LOG_TARGET,
+                        "{:#}",
+                        MmProxyError::ParseError(
```

Comment:
Why not rather return this error and log it in one place in the caller?

Reply:
It logs the error and the continues, returning the data unmodified. This will in turn be sent back to XMRig.